src/main/scala/org/broadinstitute/hail/expr/AST.scala

```diff
@@ -1631,6 +1631,191 @@ case class BinaryOp(posn: Position, lhs: AST, operation: String, rhs: AST) exten
       s.r.findFirstIn(t).isDefined
     }
 
+    case ("+", TArray(elementType)) =>
+      ((lhs.`type`, rhs.`type`, elementType): @unchecked) match {
+        case (t: TNumeric, TArray(_), TDouble) => AST.evalCompose[Any, IndexedSeq[_]](ec, lhs, rhs) { case (num, arr) =>
+          val d = DoubleNumericConversion.to(num)
+          arr.map(x => DoubleNumericConversion.to(x) + d)
+        }
```

Comment:
this also coerces the scalar once per array element, which is inefficient.  I think we should have a framework where the efficiency and abstraction are both easily specified, but that is definitely not an easy thing.